packages/components/buttons/icon-button/package.json

```diff
@@ -33,9 +33,11 @@
     "prop-types": "15.8.1"
   },
   "devDependencies": {
-    "react": "17.0.2"
+    "react": "17.0.2",
+    "react-intl": "5.24.6"
```

Comment:
I'm confused because I don't understand why we need the dependency both as _dev_ and _peer_. I mean, if we declare the dependency in the _dev_ section, we wouldn't need to add it in the _peer_ as the package would already include it on its own.

In the other hand, I guess we talked about declaring the `react-intl` package in the _peer_ section. This way, we're delegating in the consumers to provide its own version of that package instead of include it in all of the ui-kit packages.
This way we would be following the intent of the original ui-kit package (eg: `text`) by not including `react/intl` in the ui-kit packages.

Maybe I missed something along the way and there is a reason for current code proposal.

Reply:
Sorry, I could have used author's comment for that. I understand your confusion, I am confused too 😅
However, upon adding a sheer peer dependency like:
`yarn workspace @commercetools-uikit/icon-button add --peer react-intl@5.x`
I get an error:
```bash
error @commercetools-uikit/icon-button has a peerDependency on react-intl but it is not also specified in devDependencies, please add it there.
```
Therefore, (as the error msg suggests) I added it as a dev dependency too (following the pattern in all other places where `react-inlt` has already been added)

Reply:
I'm sorry but I'm still confused.
I think having `react-intl` both in _dev_ and _peer_ dependencies is not what we want.

When running the command from the package directory it doesn't log that error:
```
local:~/Development/ui-kit/packages/components/buttons/icon-button (cm-ts-presets) $ yarn add --peer react-intl@5.x
```

I'm not very familiar yet with yarn and monorepos so maybe @emmenko could help us understand this one.

Reply:
Hey guys, I thought a bit about this and I think (at least for now) it would be enough to treat these peer dependency requirements as normal dependencies.

So for example the `icon-button` package will have `react-intl` as a normal dependency (using `^` caret version for flexibility). Same for all other packages that for instance require the `text` package which has `react-intl` as a peer dependency.

Reply:
@emmenko I'm sorry, I'm having issues again 😕
```bash
yarn workspace @commercetools-uikit/flat-button add react-intl@^5.24.6
```
results in
```bash
☔️ error @commercetools-uikit/flat-button has a dependency on react-intl@^5.24.6 but the most common range in the repo is 5.24.6, the range should be set to 5.24.6
```
Would you like me rather to update all current packages using `react-intl` to use `^5.24.6` or should I simply drop the caret in the newly updated packages?

Reply:
Use `^` everywhere, thanks

Reply:
Checking again the package.json files I realized we have some packages where `react-intl` in defined in the `dependencies` group and others where it is in the `devDependencies` group and I was wondering why we have that difference but don't seem to find an explanation.

@emmenko Do you have any info about this? Should we standarize where that dependency should belong to or is it ok to have it like this?

Reply:
When it's a devDependency it's because the package is defined as a peerDependency.

Reply:
Ok. I was confused because `react-intl` will be anyway a _runtime dependency_ as it is needed by the component but I get it.

Thanks for the explanation.